docs/user/dql/functions.rst

```diff
@@ -2195,3 +2195,43 @@ Another example to show how to set custom values for the optional parameters::
     | Bond       |
     +------------+
 
+MATCH_PHRASE
```

Comment:
not sure what should be done but should we distinguish between this one and https://github.com/opensearch-project/sql/blob/main/docs/user/beyond/fulltext.rst#match-phrase-query

Reply:
Is that `match_phrase` from the legacy SQL engine?

Reply:
i believe so, i'm thinking maybe add a note to the old one, or link it after "For backward compatibility" in the new one, so users know what the difference is

Reply:
@joshuali925 that's a good point. 

Distinction between old and new SQL engines was explicitly removed as part of https://github.com/opensearch-project/sql/issues/58

I'd like to add a note to `fulltext.rst` that the document describes  the legacy SQL engine and to refer to `functions.rst` for the new engine.

Once the new engine is on par with the old one, we can update `fulltext.rst` to describe the default engine.

What do you think?